test/Libraries/RevitIntegrationTests/SelectionTests.cs

```diff
@@ -362,6 +355,126 @@ public void SelectionInSyncWithDocumentOperations_References()
+            // passed to ElementsOFCategory
+            var elementsOfCategoryNodeID = "24f225e1-8883-48c3-a8ba-773b2734336c";
+            AssertPreviewCount(elementsOfCategoryNodeID, 22);
+            for (int i = 0; i <= 21; i++)
+            {
+                var refPt = GetPreviewValueAtIndex(elementsOfCategoryNodeID, i) as ReferencePoint;
+                Assert.IsNotNull(refPt);
+            }
+
+            // ElementsOfFamilyType & Family Type, as FamilyType output passed to 
+            // ElementsOfFamilyTypes node.
+            var elementsOfFamilyTypeNodeID = "657017f1-4775-4daa-b00e-8d85392be6b5";
+            AssertPreviewCount(elementsOfFamilyTypeNodeID, 11);
+            for (int i = 0; i <= 10; i++)
+            {
+                var family = GetPreviewValueAtIndex(elementsOfFamilyTypeNodeID, i) 
+                                                        as Revit.Elements.AbstractFamilyInstance;
+                Assert.IsNotNull(family);
+            }
+
+            // ElementsOfType & ElementTypes, as ElementType output passed to ElementsOfTypes node.
+            var elementsOfTypeNodeID = "186142a6-eb54-4d39-8292-5b016a3a1c3b";
+            AssertPreviewCount(elementsOfTypeNodeID, 27);
+            for (int i = 0; i <= 26; i++)
+            {
+                var curtainGridLine = GetPreviewValueAtIndex(elementsOfTypeNodeID, i) 
+                                                            as Revit.Elements.Element;
+                Assert.IsNotNull(curtainGridLine);
+            }
+
+            // Select Model Elements
+            var selectModelElementsNodeID = "e18b4c42-71b7-4f32-b0a5-6aa89ac32aeb";
+            AssertPreviewCount(selectModelElementsNodeID, 8);
+            for (int i = 0; i <= 7; i++)
+            {
+                var wall1 = GetPreviewValueAtIndex(selectModelElementsNodeID, i) as Revit.Elements.Wall ;
+                Assert.IsNotNull(wall1);
+            }
+
+            // Select Model Element
+            var selectModelElementNode = GetPreviewValue
+                                ("137a10ad-a34f-4524-aaed-1c608e7ce8b6") as Revit.Elements.Element;
+            Assert.IsNotNull(selectModelElementNode);
+
+            // Select Face
+            var selectFaceNodeID = GetPreviewValueAtIndex
+                                        ("26942776-c0f5-4a23-9be7-95ef80809951", 0) as Surface;
+            Assert.IsNotNull(selectFaceNodeID);
+
+            // Select Faces (Below Node ID taken from Flatter to make it 1D array)
+            var selectFacesNodeID = "00a2189b-88f4-4e15-b9f0-efefa6357b37";
+            AssertPreviewCount(selectFacesNodeID, 5);
+            for (int i = 0; i <= 4; i++)
+            {
+                var surface = GetPreviewValueAtIndex(selectFacesNodeID, i) as Surface;
+                Assert.IsNotNull(surface);
+            }
+
+            // StructuralFramingTypes
+            var structuralFramingTypesNodeID = GetPreviewValue("b786fdf5-b6fa-4aab-a2fd-b40ac12b8b87");
```

Comment:
@Randy-Ma , those are Node GUID so I have appended node name to understand ID coming fomr which node